pkg/restapi/vc/operation/operations.go

```diff
@@ -731,40 +734,70 @@ func (o *Operation) createDIDUniRegistrar(keyType, signatureType, purpose string
 
 	identifier, keys, err := o.uniRegistrarClient.CreateDID(registrar.DriverURL, opts...)
 	if err != nil {
-		return "", "", "", fmt.Errorf("failed to create did doc from uni-registrar: %v", err)
+		return "", "", fmt.Errorf("failed to create did doc from uni-registrar: %v", err)
 	}
 
 	// TODO remove check when vendors supporting addKeys feature
 	if strings.Contains(identifier, "did:trustbloc") {
 		for _, v := range keys {
 			if strings.Contains(v.ID, "#"+selectedKeyID) {
-				return identifier, v.ID, "", nil
+				return identifier, v.ID, nil
 			}
 		}
 
-		return "", "", "", fmt.Errorf("selected key not found %s", selectedKeyID)
+		return "", "", fmt.Errorf("selected key not found %s", selectedKeyID)
 	}
 
 	if strings.Contains(identifier, "did:v1") {
 		for _, v := range keys {
 			for _, p := range v.Purpose {
 				if purpose == p {
-					return identifier, v.ID, v.PrivateKeyBase58, nil
+					err = o.importKey(v.ID, kms.ED25519Type, base58.Decode(v.PrivateKeyBase58))
+					if err != nil {
+						return "", "", err
+					}
+
+					return identifier, v.ID, nil
 				}
 			}
 		}
 
-		return "", "", "", fmt.Errorf("did:v1 - not able to find key with purpose %s", purpose)
+		return "", "", fmt.Errorf("did:v1 - not able to find key with purpose %s", purpose)
 	}
 
 	// vendors not supporting addKeys feature.
 	// return first key public and private
 	// TODO remove when vendors supporting addKeys feature
-	return identifier, keys[0].ID, keys[0].PrivateKeyBase58, nil
+	err = o.importKey(keys[0].ID, kms.ED25519Type, base58.Decode(keys[0].PrivateKeyBase58))
+	if err != nil {
+		return "", "", err
+	}
+
+	return identifier, keys[0].ID, nil
+}
+
+func (o *Operation) importKey(keyID string, keyType kms.KeyType, privateKeyBytes []byte) error {
+	split := strings.Split(keyID, "#")
+
+	var privKey interface{}
+
+	switch keyType {
+	case kms.ED25519Type:
+		privKey = ed25519.PrivateKey(privateKeyBytes)
+	default:
+		return fmt.Errorf("import key type not supported %s", keyType)
+	}
+
+	_, _, err := o.kms.ImportPrivateKey(privKey,
```

Comment:
Are we going to have key id collisions? For example in Sidetree protocol key id can be an arbitrary name.

Reply:
we don't control the id

Reply:
but collisions is  possible

Reply:
we need to revisit

Reply:
hopefully KMS will reject it, otherwise we may overwrite someone's key

Reply:
Since KMS will reject import if key already exists we should be safe here